### DISCUSSION TITLE
Android foreground service without thread

### DISCUSSION THREAD

**Discussion by ebrahimkarimi** [socketio/socket.io#4850] [Q&A]
is [this](https://stackoverflow.com/q/77272835/7337499) a bug?

**Comment by darrachequesne** [comment]
Hi! It seems ProGuard needs some [specific configuration with SignalR](https://stackoverflow.com/questions/51610646/what-are-the-proguard-rules-for-signalr-android-client) (a library kind of similar to Socket.IO), could you please check with:

```
// proguard-rules.pro

 -keep class io.socket.** { *; }
 -keep interface io.socket.** { *; }
```
